cmd/shcd.go

```diff
@@ -290,12 +314,109 @@ func (id Id) GenerateXname() (xn string) {
 
 		xn = x.String()
 
+	} else if strings.HasPrefix(id.CommonName, "ncn-") {
+
+		// Convert the rack to a string
+		cabString := id.Location.Rack
+
+		// Strip the "x"
+		_, cabNum := utf8.DecodeRuneInString(cabString)
+
+		// Convert to an int
+		cabinet, err := strconv.Atoi(cabString[cabNum:])
+
+		if err != nil {
+			log.Fatalln(err)
+		}
+
+		// Strip the u
+		i := strings.TrimPrefix(id.Location.Elevation, "u")
+
+		// Check if this is a dense 4 node chassis or dual node chassis as additional logic is needed for these to find the slot number
+		if strings.HasSuffix(i, "L") || strings.HasSuffix(i, "R") {
+			// Dense 4 node chassis - Gigabyte or Intel chassis --
+			// The BMC ordinal for the nodes BMC is derived from the NID of the node, by applying a modulo of 4 plus 1
+			if id.Vendor == "gigabyte" || id.Vendor == "intel" {
+
+				i = strings.TrimSuffix(i, "L")
+				i = strings.TrimSuffix(i, "R")
+
+				slot, err := strconv.Atoi(i)
+
+				if err != nil {
+					log.Fatalln(err)
+				}
+
+				bmcOrdinal = (slot % 4) + 1
+
+				// Dual node chassis - Apollo 6500 XL645D -- L == b1, R == b2
+			} else if id.Vendor == "hpe" {
```

Comment:
Do we have hpe anywhere as a global constant?

Reply:
No, but good idea..